### PR TITLE
feat: add search filter

### PR DESCRIPTION
**Pull request by algoritmi4**

### CODE REVIEW

File: src/pages/home/HomePage.tsx [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import { useAppSelector } from '@/shared/model';
 import { EventsList } from '@/widgets/EventsList';
-import {FC} from 'react';
+import { ReactElement } from 'react';
 
-const HomePage: FC = () => {
-  const { data: events = {results: []}, isLoading } = useGetEventsQuery();
+export function HomePage(): ReactElement {
+  const searchFilter = useAppSelector(state => state.globalFilter.search);
+  const { data: events = {results: []}, isLoading, isError } = useGetEventsQuery(searchFilter && searchFilter);
```
Comment (sakalouski-alex) [inline src/pages/home/HomePage.tsx]: Why can't we use simple `useGetEventsQuery(searchFilter)`  here?

File: src/pages/home/HomePage.tsx [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+
+  isError && console.log('Ошибка при получении ивентов');
```
Comment (sakalouski-alex) [inline src/pages/home/HomePage.tsx]: I'm not sure if we want to stay what console.log

File: src/entities/event/api/eventApi.ts [hunk 3]
```diff
@@ -10,13 +10,19 @@ interface ApiResponse<T> {
...
-      query: () => ({
+    getEvents: build.query<IApiResponse<IEvent[]>, string | null>({
+      query: (filter) => (filter ? {
         url: '/events/',
-        method: 'GET'
+        method: 'GET',
+        params: {
+          search: filter
+        }
+      } : {
```
Comment (sakalouski-alex) [inline src/entities/event/api/eventApi.ts]: For now we are only using the serach filter, but this endpoint can be used with multiple filters.

```suggestion
// Add EventFilter type
EventFilter {
search?: string
}
...
// Use it as endpoint arg
....
query: (filter: EventFilter) => ({
        url: '/events/',
        method: 'GET',
        params: {
          search: filter.search
        }
    ....
```

We can use empty default values for filter parameters and the backend will ignore them, am I right @jakkemomo ?

File: src/pages/home/HomePage.tsx (hunk 1 above)
Comment (sakalouski-alex) [reply src/pages/home/HomePage.tsx]: Or, if you apply my suggestion above, this call would be replaced with `useGetEventsQuery({search: searchFilter})`

Comment (sakalouski-alex) [reply src/pages/home/HomePage.tsx]: In this case, when we add new filters, this call will remain the same, unchanged

File: src/entities/event/api/eventApi.ts (hunk 3 above)
Comment (sakalouski-alex) [reply src/entities/event/api/eventApi.ts]: > We can use empty default values for filter parameters and the backend will ignore them, am I right @jakkemomo ?

I checked in the backend, this is correct, at least for the search argument.

File: src/pages/home/HomePage.tsx (hunk 2 above)
Comment (algoritmi4) [reply src/pages/home/HomePage.tsx]: what do u mean? like this - console.log(`Ошибка при получении ивентов - ${error}`)?

Comment (sakalouski-alex) [reply src/pages/home/HomePage.tsx]: Yeah, that's better.

File: src/app/rootReducer.ts [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
+import {combineReducers} from '@reduxjs/toolkit';
+import {baseApi, jwtApi} from '@/shared/api';
 import {SessionSlice} from "@/entities/session/model/slice";
-import {registerFormSlice} from '@/features/authentication/registration/model/formState'
+import globalFilterReducer from '@/features/globalFilter/model/GlobalFilterSlice';
+import {registerFormSlice} from '@/features/authentication/registration/model/formState';
 
 export const rootReducer = combineReducers({
   session: SessionSlice.reducer,
+  globalFilter: globalFilterReducer,
```
Comment (snurnisyan) [inline src/app/rootReducer.ts]: I would suggest to rename globalFilter to searchFilter because you have changed the name of that component

Comment (algoritmi4) [reply src/app/rootReducer.ts]: globalFilter is a slice for all filters including filters from drop-down menu. i have no maquette for drop-down, thats why i cant add more filters in this slice yet

Comment (snurnisyan) [reply src/app/rootReducer.ts]: but still this filter is for search, why don't call it search? global filter doesn't describe that this filter is used for searching an event

**Pull request merged**